bigframes/session/__init__.py

```diff
@@ -495,14 +488,26 @@ def _get_snapshot_sql_and_primary_key(
 
         job_config = bigquery.QueryJobConfig()
         job_config.labels["bigframes-api"] = api_name
-        current_timestamp = list(
-            self.bqclient.query(
-                "SELECT CURRENT_TIMESTAMP() AS `current_timestamp`",
-                job_config=job_config,
-            ).result()
-        )[0][0]
+        if use_cache:
+            if table_ref in self._df_snapshot.keys():
+                snapshot_timestamp = self._df_snapshot[table_ref]
+            else:
+                snapshot_timestamp = list(
+                    self.bqclient.query(
+                        "SELECT CURRENT_TIMESTAMP() AS `current_timestamp`",
+                        job_config=job_config,
+                    ).result()
+                )[0][0]
+                self._df_snapshot[table_ref] = snapshot_timestamp
+        else:
+            snapshot_timestamp = list(
+                self.bqclient.query(
+                    "SELECT CURRENT_TIMESTAMP() AS `current_timestamp`",
+                    job_config=job_config,
+                ).result()
+            )[0][0]
```

Comment:
I think we want to update the snapshot timestamp in this code path too. I believe that's how BQ works where it'll still populate the cache but not read from it.

Reply:
Done.